ci-bench/src/bench_lib/benchmark.rs

```diff
@@ -0,0 +1,158 @@
+use crate::cli::Side;
+use itertools::Itertools;
+use std::collections::HashSet;
+use std::io::{Read, Write};
+
+/// A collection of benchmark scenarios
+pub struct BenchmarkCollection {
```

Comment:
What is the value of this type? It seems we could just pass a `Vec<Benchmark>` around which would be more straightforward.

Reply:
The type is meant to encode the invariant that the collection of benchmarks has been successfully validated. I was hoping it would make the code easier to read, but I can remove it if people have a clear preference for a plain `Vec`.

Reply:
Currently I think we have only one place that create the `Vec<Benchmark>`, so for now it feels like overkill to add a wrapper for this -- instead we can just do the validation in that function.

Reply:
Done 😉